### PR TITLE
network: tc: introduce ETS

### LINKED ISSUE #15264: network: tc: add sch_ets qdisc
In Linux 5.6 https://github.com/torvalds/linux/commit/dcc68b4d8084e1ac9af0d4022d6b1aff6a139a33  sch_ets was added - [802.1Qaz – Enhanced Transmission Selection](https://1.ieee802.org/dcb/802-1qaz/).

### PR DESCRIPTION
**Pull request by yuwata**
Closes #15264.

Not tested yet. Please don't merge this.

cc @k0ste and @ssahani.

### CODE REVIEW

File: man/systemd.network.xml [hunk 1]
```diff
@@ -2962,6 +2962,73 @@
...
+    <title>[EnhancedTransmissionSelection] Section Options</title>
+    <para>The <literal>[EnhancedTransmissionSelection]</literal> section manages the queueing discipline (qdisc) of
+    Enhanced Transmission Selection (ETS).</para>
+
+    <variablelist class='network-directives'>
+      <xi:include href="tc.xml" xpointer="qdisc-parent" />
+      <xi:include href="tc.xml" xpointer="qdisc-handle" />
+
+      <varlistentry>
+        <term><varname>Band=</varname></term>
```
Comment (poettering) [inline man/systemd.network.xml]: shouldn't this be in plural? Bands=?

File: man/systemd.network.xml [hunk 2]
```diff
@@ -2962,6 +2962,73 @@
...
+        <listitem>
+          <para>Specifies the number of bands. An unsigned integer ranges 1 to 16. This value has to be
+          at least large enough to cover the strict bands specified through the <varname>Strict=</varname>
+          and bandwidth-sharing bands specified in <varname>Quanta=</varname>.</para>
+        </listitem>
+      </varlistentry>
+
+      <varlistentry>
+        <term><varname>Strict=</varname></term>
```
Comment (poettering) [inline man/systemd.network.xml]: if these are "bands" too maybe call this StrictBands=

File: man/systemd.network.xml [hunk 3]
```diff
@@ -2962,6 +2962,73 @@
...
+          <para>Specifies the number of bands that should be created in strict mode. An unsigned integer
+          ranges 1 to 16.</para>
+        </listitem>
+      </varlistentry>
+
+      <varlistentry>
+        <term><varname>Quanta=</varname></term>
+        <listitem>
+          <para>Specifies the white-space separated list of quantum used in band-sharing bands. Each
+          unsigned integer ranges 1 to 4294967294.</para>
```
Comment (poettering) [inline man/systemd.network.xml]: what's the unit here?

**Comment by ssahani** [comment]
I tested it. Works fine.

```
root@arch /lib/systemd/network
❯ tc qdisc
qdisc noqueue 0: dev lo root refcnt 2
qdisc ets 1: dev ens192 root refcnt 5 bands 8 quanta 1514 1514 1514 1514 1514 1514 1514 1514 priomap 7 6 5 4 3 7 7 7 7 7 7 7 7 7 7 7
qdisc ets 3a: dev dummy98 root refcnt 2 bands 10 strict 3 quanta 1514 1514 1514 1514 1514 1514 1514 priomap 9 9 9 9 9 9 9 9 9 9 9 9 9 9 9 9

root@arch /lib/systemd/network
❯ cat dummy.network
[Match]
Name=dummy98

[Network]
IPv6AcceptRA=no
Address=10.1.2.3/16

[EnhancedTransmissionSelection]
Parent=root
Handle=3a
Band=10
Strict=3
Quanta=1 2 3 4 5
PriorityMap=3 4 5 6 7

[EnhancedTransmissionSelectionClass]
Parent=root
ClassId=3a:30
Quantum=4
root@arch /lib/systemd/network
```

**Comment by yuwata** [comment]
Oh, thank you for testing.

**Comment by yuwata** [comment]
Updated. Now configs are renamed
- `Band=` -> `Bands=`
- `Strict=` -> `StrictBands=`
- `Quantum=` -> `QuantumBytes=`

And parser errors are fixed.
@ssahani Could you test this PR?

@poettering How should we name the setting `Quanta=`? It is a list of bytes.

**Comment by ssahani** [comment]
yes this one OK
```
❯ tc qdisc
qdisc ets 3a: dev dummy98 root refcnt 2 bands 10 strict 3 quanta 1 2 3 4 5 1514 1514 priomap 3 4 5 6 7 9 9 9 9 9 9 9 9 9 9 9
```

**Comment by yuwata** [comment]
`Quanta=` is simply renamed to `QunatumBytes=` and the document is updated to mention it takes a list.

**Pull request merged**